### PR TITLE
#165020220 Social sharing

### PR DESCRIPTION
**Pull request by fridolinho**
## What does this PR do?
Add social sharing
## Description of Task to be completed?
- Validate if shared article exists
- Share using email, facebook and twitter
## How should this be manually tested?
1. Run these commands
     - clone https://github.com/andela/ah-kgl-avengers-backend.git
     - git checkout ft-social-sharing-165020220
     - npm install
     - npm test
2. 
     - send a get request on  localhost:3000/api/v1/articles/:slug/facebook-share
                                             localhost:3000/api/v1/articles/:slug/twitter-share
                                             localhost:3000/api/v1/articles/:slug/email-share
         slug is an article unique identifier, get it from existing article.
## What are the relevant pivotal tracker stories?
[#165020220 Social sharing](https://www.pivotaltracker.com/story/show/165020220)
## Screenshots (if appropriate)
N/A
## Questions:
N/A

### CODE REVIEW

File: controllers/articles.js [hunk 1]
```diff
@@ -510,6 +511,36 @@ const articles = {
...
       }
     }
+  },
+
+  fbShare: async (req, res) => {
+    const { slug } = req.params;
+    const link = `${process.env.SERVER_ADDRESS}/articles/${slug}`;
+    res.status(200).send({
+      status: res.statusCode,
+      message: 'facebook share',
```
Comment (aaronsekisambu) [inline controllers/articles.js]: @fridolinho I think the message here is not descriptive. Could you please work on it.

File: controllers/articles.js [hunk 2]
```diff
@@ -510,6 +511,36 @@ const articles = {
...
+      url: link
+    });
+    open(`https:www.facebook.com/sharer/sharer.php?u=https://${link}`);
+  },
+
+  twitterShare: async (req, res) => {
+    const { body } = req.article;
+    res.status(200).send({
+      status: res.statusCode,
+      message: 'twitter share'
```
Comment (aaronsekisambu) [inline controllers/articles.js]: @fridolinho same applies to this one. It's also not descriptive.

File: controllers/articles.js [hunk 3]
```diff
@@ -510,6 +511,36 @@ const articles = {
...
+    });
+    open(`https://twitter.com/intent/tweet?text=${body}`);
+  },
+
+  emailShare: async (req, res) => {
+    const { title, body } = req.article;
+    res.status(200).send({
+      status: res.statusCode,
+      message: 'email share',
```
Comment (aaronsekisambu) [inline controllers/articles.js]: @fridolinho and this one.

**Review by aaronsekisambu — CHANGES_REQUESTED** [review]

**Review by niyobobo — DISMISSED** [review]
Better if you return a descriptive message rather than `facebook share, twitter share,......`

**Comment by fridolinho** [comment]
Resolved everything

**Review by aaronsekisambu — DISMISSED** [review]

**Review by shemaeric — DISMISSED** [review]
LGTM

**Review by daud1 — APPROVED** [review]
👍

**Pull request merged**